epic_capybara/cli/bara.py

```diff
@@ -54,14 +54,21 @@ def bara(files, match, unmatch, serve):
 
     for _file in files:
         tree = uproot.open(_file)["events"]
+
+        sort_by_evtnum = None
+        if "EventHeader/EventHeader.eventNumber" in tree.keys(recursive=True):
+            evtnum = tree["EventHeader/EventHeader.eventNumber"].array()
+            sort_by_evtnum = ak.argsort(ak.flatten(evtnum))
+
         keys = [
             key for key in tree.keys(recursive=True)
             if not key.startswith("PARAMETERS")
             and len(tree[key].branches) == 0
             and match_filter(key, match, unmatch)
         ]
+
         for key in keys:
-            arr.setdefault(key, {})[_file] = tree[key].array()
+            arr.setdefault(key, {})[_file] = tree[key].array() if sort_by_evtnum is None else tree[key].array()[sort_by_evtnum]
```

Comment:
```suggestion
            arr.setdefault(key, {})[_file] = tree[key].array()
            if sort_by_evtnum is not None:
                arr[key][_file] = arr[key][_file][sort_by_evtnum]
```

Reply:
Does this actually work? I'm never quite sure when something in python is just a view and when it creates a copy.

Reply:
Yes, this will work, it's just indexing into nested dictionaries, which gives an lvalue. And, if you prefer, I was contemplating proposing this first:
```suggestion
            val = tree[key].array()
            if sort_by_evtnum is not None:
                val = val[sort_by_evtnum]
            arr.setdefault(key, {})[_file] = val
```

Reply:
No preference on my side. In more familiar with c++ style preferences.